translate_input.rb

```
@@ -0,0 +1,31 @@
+# Users enter move in algebraic notation (e.g "e2 e4")
+# The movement code needs these in 0 based coordinates (e.g "41 43")
+module TranslateInput
+  class << self
+    FILES = %w[a b c d e f g h].freeze
+    MOVEMENT_COMMAND_LENGTH = 5
+
+    def algebraic_to_0_based(movement_string)
+      return nil if movement_string.length != MOVEMENT_COMMAND_LENGTH
+      zero_based =
+        zero_base_all_indices(rank_letters_to_numbers(movement_string))
+      return nil if zero_based.flatten.any?(&:negative?)
```

its a tough design choice, but should this method return nil if an invalid argument was provided? Thats kind of a silent error